lightning/src/ln/channel.rs

```diff
@@ -3058,6 +3094,31 @@ impl<Signer: Sign> Channel<Signer> {
 			panic!("Cannot update fee while peer is disconnected/we're awaiting a monitor update (ChannelManager should have caught this)");
 		}
 
+		// Before proposing a feerate update, check that we can actually afford the new fee.
+		let inbound_stats = self.get_inbound_pending_htlc_stats(Some(feerate_per_kw));
+		let outbound_stats = self.get_outbound_pending_htlc_stats(Some(feerate_per_kw));
+		let keys = if let Ok(keys) = self.build_holder_transaction_keys(self.cur_holder_commitment_transaction_number) { keys } else { return None; };
+		let commitment_stats = self.build_commitment_transaction(self.cur_holder_commitment_transaction_number, &keys, true, true, logger);
+		let buffer_fee_msat = Channel::<Signer>::commit_tx_fee_sat(feerate_per_kw, commitment_stats.num_nondust_htlcs + CONCURRENT_INBOUND_HTLC_FEE_BUFFER as usize) * 1000;
```

Comment:
I believe since we're subtracting the holding cell balance below, the number of holding cell HTLCs should also be factored into `buffer_fee_msat`

Reply:
Okay I added a `holding_cell_htlcs_count` to HTLCStats but thinking more I think this approach is over-conservative. As any holding cell HTLC is expected to be committed in the same sequence we could instead delay the holding cell accounting when it's actually _sent_, assuming we do it at pending update_fee. This is not what we're doing in `send_htlc` rn because `next_local_commit_tx` use the maybe-legacy fee but I think it wouldn't be that much work once we re-use the fee from `build_commitment_transaction`.

Note, the spec is pretty loose on how to account holding_cell/pending HTLCs.